extensions/obsidian/src/createnote.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  List,
+  ActionPanel,
+  getPreferenceValues,
+  PushAction,
+  Form,
+  SubmitFormAction,
+  useNavigation,
+  showToast,
+  ToastStyle,
+  AlertOptions,
+  confirmAlert,
+  Icon,
+} from "@raycast/api";
+import fs from "fs";
+import path from "path";
+
+interface Vault {
+  name: string;
+  key: string;
+}
+
+interface Preferences {
+  vaultPath: string;
+  prefPath: string;
+  prefTag: string;
+  tags: string;
+  primaryAction: string;
+}
+
+interface FormValue {
+  path: string;
+  name: string;
+  content: string;
+  tags: Array<string>;
+}
+
+function prefVaults() {
+  const pref: Preferences = getPreferenceValues();
+  const vaultString = pref.vaultPath;
+  return vaultString
+    .split(",")
+    .map((vault) => ({ name: vault.trim(), key: vault.trim() }))
+    .filter((vault) => !!vault);
+}
+
+function prefPath(): string {
+  const pref: Preferences = getPreferenceValues();
+  const prefPath = pref.prefPath;
+  if (prefPath) {
+    return prefPath;
+  }
+  return "";
+}
+
+function prefTag(): Array<string> {
+  const pref: Preferences = getPreferenceValues();
+  const prefTag = pref.prefTag;
+  if (prefTag) {
+    return [prefTag];
+  }
+  return [];
+}
+
+function tags() {
+  const pref: Preferences = getPreferenceValues();
+  const tagsString = pref.tags;
+  const prefTag = pref.prefTag;
+  if (!tagsString) {
+    return [{ name: prefTag, key: prefTag }];
+  }
+  const tags = tagsString
+    .split(",")
+    .map((tag) => ({ name: tag.trim(), key: tag.trim() }))
+    .filter((tag) => !!tag);
+  tags.push({ name: prefTag, key: prefTag });
+  return tags;
+}
+
+function NoteForm(props: { vaultPath: string }) {
+  const vaultPath = props.vaultPath;
+  const { pop } = useNavigation();
+
+  async function writeNote(notePath: string, name: string, content: string) {
+    if (fs.existsSync(path.join(notePath, name + ".md"))) {
+      const options: AlertOptions = {
+        title: "Override note",
+        message: 'Are you sure you want to override the note: "' + name + '"?',
+        icon: Icon.ExclamationMark,
+      };
+      if (await !confirmAlert(options)) {
+        return;
+      }
+    }
+
+    try {
+      fs.mkdirSync(notePath, { recursive: true });
+      fs.writeFileSync(path.join(notePath, name + ".md"), content);
+      showToast(ToastStyle.Success, "Created new note");
+      pop();
+    } catch {
+      showToast(ToastStyle.Failure, "Something went wrong. Maybe your vault, path or filename is not valid.");
+    }
+  }
```

Comment:
```suggestion
  async function writeNote(notePath: string, name: string, content: string) {
    if (fs.existsSync(path.join(notePath, name + ".md"))) {
      const options: AlertOptions = {
        title: "Override note",
        message: 'Are you sure you want to override the note: "' + name + '"?',
        icon: Icon.ExclamationMark,
      };
      if (await confirmAlert(options)) {
        try {
          fs.mkdirSync(notePath, { recursive: true });
          fs.writeFileSync(path.join(notePath, name + ".md"), content);
          showToast(ToastStyle.Success, "Created new note");
          pop();
        } catch {
          showToast(ToastStyle.Failure, "Something went wrong. Maybe your vault, path or filename is not valid.");
        }
      }
      }
    }
```
You should switch the `confirmAlert`, right now It's popping to root and confirm override. If you do it the suggested way, it will stay in the same window and let the user cancel and change input.

Reply:
Thank you, added these changes in the new commit